pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -635,14 +717,19 @@ func (r *ReconcileBareMetalHost) manageHostPower(prov provisioner.Provisioner, i
 	if provResult.Dirty {
 		info.postSaveCallbacks = append(info.postSaveCallbacks, func() {
 			metricLabels := hostMetricLabels(info.request)
-			if info.host.Spec.Online {
+			if desiredPowerOnState {
 				metricLabels[labelPowerOnOff] = "on"
 			} else {
 				metricLabels[labelPowerOnOff] = "off"
 			}
 			powerChangeAttempts.With(metricLabels).Inc()
 		})
 		info.host.ClearError()
+		if desiredPowerOnState {
+			info.host.RecordPoweredOn()
+		} else {
+			info.host.Status.Provisioning.PoweredOnAt = nil
```

Comment:
I'd expect we'd want to do this once `prov.UpdateHardwareState()` tells us that the power state is off. Here we're doing it as soon as we've asked Ironic to turn the power off, without waiting to see whether it is successful.